covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -0,0 +1,328 @@
+            case "cumulativePerCapita":
+                this.setState({cumulativePerCapita: data})
+                break;
+        }
+    }
+
+    /**
+     * The chart takes a labelList. Which is a list of strings that will be shown on top of each bar.
+     * This will pre-generate those strings.
+     * For example, if the chart is a percent chart, we want to generate +800 (12%).
+     * @param value
+     * @param absolute
+     * @param population
+     */
+    getLabelListLabel = (value, absolute, population) => {
+        if (absolute && population) return numberWithCommas(absolute) + ' / ' + numberWithCommas(population)
+        else if (absolute && value) return "+" + numberWithCommas(absolute) + " (" + ( value) + "%)"
+        else return numberWithCommas(value)
+    }
+
+    /**
+     * Converts an ISO date to English date.
+     * For example, 2020-01-01 is converted to January 1st, 2020.
+     * @param date
+     */
+    prettifyDate = (date) => {
+        if (date.toLowerCase() === "latest") return "Daily"
+        const months = {
+            1: "January",
+            2: "February",
+            3: "March",
+            4: "April",
+            5: "May",
+            6: "June",
+            7: "July",
+            8: "August",
+            9: "September",
+            10: "October",
+            11: "November",
+            12: "December"
+        }
+        const tempDate = new Date(date)
+        let day = tempDate.getDate() + 1
+        let month = tempDate.getMonth() + 1
+        let year = tempDate.getFullYear()
+
+        return `${months[month]} ${day}th, ${year}`
+    }
+
+
+    /**
+     * Given a hashmap of data, convert it to a list dataHolder objects.
+     * Also sorts the list so that the chart shows the regions from highest to lowest.
+     * This is necessary as Chart.jsx only know how to interpret a dataHolder list.
+     * @param data
+     */
+    jsonToArray(data): dataHolder[]{
+        let dataAsList: any = []
+        let dcids: string[] = Object.keys(data['value'])
+
+        dcids.forEach(dcid => {
+            let regionName: string = this.state.dcidMap[dcid]
+            let value: number = Math.round(data?.value[dcid] * 100000) / 100000
+            let absolute: number = data?.absolute?.[dcid]
+            let population: number = data?.population?.[dcid]
+
+            dataAsList.push({
+                regionName: regionName,
+                value: value,
+                absolute: absolute,
+                population: population,
+                dcid: dcid,
+                labelListLabel: this.getLabelListLabel(value, absolute, population)
+            })
+        })
+        return dataAsList.sort((a, b) => b.value - a.value).slice(0, this.state.show)
+    }
+
+    render() {
+        return (
+            <div className={"container"}>
+                <SideNav handleScrollOnRef={this.handleScrollOnRef} date={this.state.date}/>
+                <div className={"main-content"}>
```

Comment:
can you try to indent this section of code, ideally each prop starts with a new line

Reply:
Another idea is to have a config file to hold all these information and you just need a for loop. 
The config file could be a json file

Reply:
There is a json file holding this info now, and I map through it to generate the components